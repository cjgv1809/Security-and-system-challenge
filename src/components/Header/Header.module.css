.header {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 1rem;
  padding: 1.25rem;
  width: 100%;
}

.profileImage {
  width: 110px;
  height: 110px;
  border-radius: 50%;
  border: 3px solid var(--color-primary);
  object-fit: cover;
}

.name {
  font-size: 2rem;
  color: var(--color-active-text);
}

@media (max-width: 600px) {
  .header {
    flex-direction: column;
    text-align: center;
  }
}
