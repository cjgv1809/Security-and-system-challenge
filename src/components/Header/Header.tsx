"use client";

import React from "react";
import Image from "next/image";
import { motion } from "framer-motion";
import styles from "./Header.module.css";

type HeaderProps = {
  name: string;
  imageUrl: string;
};

const Header: React.FC<HeaderProps> = ({ name, imageUrl }) => {
  return (
    <header className={styles.header}>
      <Image
        src={
          "https://images.unsplash.com/photo-1619996107119-31de3403d467?w=500&auto=format&fit=crop&q=60&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MTZ8fHByb2ZpbGUlMjBpbWFnZXN8ZW58MHx8MHx8fDA%3D"
        }
        alt={name}
        className={styles.profileImage}
        width={100}
        height={100}
        quality={95}
        priority
      />
      <motion.h1
        className={styles.name}
        initial={{ opacity: 0, y: -20 }}
        animate={{ opacity: 1, y: 0 }}
        transition={{ duration: 0.5, delay: 0.2 }}
      >
        {name}
      </motion.h1>
    </header>
  );
};

export default Header;
