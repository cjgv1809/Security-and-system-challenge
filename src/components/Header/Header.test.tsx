import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "./Header";

describe("Header", () => {
  it("renders the header with the name", () => {
    const name = "John Doe";
    render(<Header name={name} imageUrl="/next.svg" />);

    expect(screen.getByText(name)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const name = "John Doe";
    const { container } = render(<Header name={name} imageUrl="/next.svg" />);

    expect(container).toMatchSnapshot();
  });
});
