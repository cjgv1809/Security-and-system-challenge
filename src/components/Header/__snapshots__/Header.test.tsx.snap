// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <img
      alt="John Doe"
      class="profileImage"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="100"
      src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1619996107119-31de3403d467%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MTZ8fHByb2ZpbGUlMjBpbWFnZXN8ZW58MHx8MHx8fDA%253D&w=256&q=95"
      srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1619996107119-31de3403d467%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MTZ8fHByb2ZpbGUlMjBpbWFnZXN8ZW58MHx8MHx8fDA%253D&w=128&q=95 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1619996107119-31de3403d467%3Fw%3D500%26auto%3Dformat%26fit%3Dcrop%26q%3D60%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MTZ8fHByb2ZpbGUlMjBpbWFnZXN8ZW58MHx8MHx8fDA%253D&w=256&q=95 2x"
      style="color: transparent;"
      width="100"
    />
    <h1
      class="name"
      style="opacity: 0; transform: translateY(-20px) translateZ(0);"
    >
      John Doe
    </h1>
  </header>
</div>
`;
